### PR TITLE
Add downscope_token() to client

### PR DESCRIPTION
**Pull request by anthonywee**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @anthonywee has signed the CLA. Thanks for the pull request!

**Comment by anthonywee** [comment]
This is a PR to compare 2 options for allowing for retries of a failed downscope_token() request. The other PR is #246

File: boxsdk/client/client.py [hunk 1]
```diff
@@ -438,6 +439,46 @@ def create_user(self, name, login=None, **user_attributes):
...
+        :type additional_data:
+            `dict`
+        :return:
+            The response for the downscope token request.
+        :rtype:
+            :class:`TokenResponse`
+        """
+        print "new downscope token method!"
+        url = '{base_auth_url}/token'.format(base_auth_url=API.OAUTH2_API_URL)
+        access_token = self.auth.access_token[:-3] + "wee"
```
Comment (jmoldow) [inline boxsdk/client/client.py]: Remove this.

File: boxsdk/client/client.py [hunk 2]
```diff
@@ -438,6 +439,46 @@ def create_user(self, name, login=None, **user_attributes):
...
+        :param additional_data:
+            (Optional) Key value pairs which can be used to add/update the default data values in the request.
+        :type additional_data:
+            `dict`
+        :return:
+            The response for the downscope token request.
+        :rtype:
+            :class:`TokenResponse`
+        """
+        print "new downscope token method!"
```
Comment (jmoldow) [inline boxsdk/client/client.py]: Remove print statements.

**Comment by jmoldow** [comment]
I think this is the best approach, though I think #246 is a good change in its own right.

Can you remove `downscope_token` from `OAuth2`, and also update the HISTORY file?

**Pull request merged**